bases/shared/instagoric-server/server.js

```diff
@@ -772,6 +775,118 @@ faucetapp.get('/transaction-status/:txhash', (req, res) => {
 });
 
+publicapp.get('/install-bundle', async (req, res) => {
+  res.send(`
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>File Upload</title>
+</head>
+<body>
+
+    <h1>Install Bundle and Submit Core Eval Proposal</h1>
+
+    <form id="uploadForm" enctype="multipart/form-data">
+        <input type="file" name="file" id="fileInput" multiple required>
+        <br><br>
+        <button type="submit">Upload and Submit Proposal</button>
+    </form>
+
+    <script>
+        document.getElementById('uploadForm').addEventListener('submit', function(event) {
+            event.preventDefault(); // Prevent the default form submission
+
+            const formData = new FormData();
+            const fileInput = document.getElementById('fileInput');
+            
+            if (fileInput.files.length === 0) {
+                alert("Please select a file.");
+                return;
+            }
+
+            // Loop through all selected files and append them to FormData
+            for (let i = 0; i < fileInput.files.length; i++) {
+                formData.append('files[]', fileInput.files[i]);
+            }
+
+            fetch('/install-bundle', {
+                method: 'POST',
+                body: formData
+            })
+            .then(data => {
+                alert('File uploaded successfully!');
+                console.log(data);
+            })
+            .catch(error => {
+                alert('Failed to upload file.');
+                console.error('Error:', error);
+            });
+        });
+    </script>
+
+</body>
+</html>
+  `);
+});
+
+publicapp.post('/install-bundle', upload.array('files[]'), async (req, res) => {
+  // Check if there are files in the request
+  const files = req.files;
+  if (!Array.isArray(files) || files.length === 0) {
+    res.status(400).send('No files uploaded');
+    return;
+  }
+
+  // Iterate through all files and process each one
+  for (const file of files) {
+    const fileName = file.originalname;
+    const bundle = file.filename;
+
+    if (fileName.startsWith('b1-')) {
+      try {
+        const result = await $`\
+        ${agBinary} tx swingset install-bundle --compress "@uploads/${bundle}" \
+        --from ${FAUCET_KEYNAME} --keyring-backend=test --keyring-dir=${agoricHome} --gas=auto \
+        --chain-id=${chainId} --yes
+      `;
+
+        if (result.exitCode !== 0) {
+          res
+            .status(500)
+            .send(`Error processing file ${bundle}: ${result.stderr}`);
+          return;
+        }
+      } catch (error) {
+        res
+          .status(500)
+          .send(`Error processing file ${bundle}: ${error.message}`);
+        return;
+      }
+    }
+  }
+
+  const proposalPermitFile = files
+    .filter(file => file.originalname.endsWith('permit.json'))
+    .map(file => file.filename)[0];
```

Comment:
If there are multiple .js and permit.json files, then it'll break as it is picking only [0]th index.

Also can we add validation if correct pair of permit.json and .js file is provided?

@toliaqat Thoughts?